extension/searching.js

```diff
@@ -12,6 +12,49 @@ this.searching = (function() {
     return searchUrl.href;
   };
 
+  exports.ddgEntitySearch = async function(query) {
+    const response = await fetch(
+      `https://api.duckduckgo.com/?q=${query}&format=json&pretty=1&skip_disambig=1`
+    );
+    const ddgData = await response.json();
+    if (!ddgData.AbstractText) {
+      // the response from DDG was null, and there was no matching Instant Answer result
+      return {};
+    }
+    const cardData = (({
+      Heading,
+      AbstractText,
+      AbstractSource,
+      AbstractURL,
+      Image,
+    }) => ({ Heading, AbstractText, AbstractSource, AbstractURL, Image }))(
+      ddgData
+    );
+    return cardData;
+  };
+
+  exports.ddgBangSearchUrl = async function(query, service) {
+    const SERVICE_BANG_MAP = {
+      "google slides": "gslides",
+      "google docs": "gd",
+      "google scholar": "googlescholar",
+      calendar: "gcal",
+      "google calendar": "gcal",
+      "google drive": "drive",
+      spotify: "spotify",
+      goodreads: "goodreads",
+      mdn: "mdn",
+      coursera: "coursera",
+    };
+    const bang = SERVICE_BANG_MAP[service.toLowerCase()];
+    const response = await fetch(
+      `https://api.duckduckgo.com/?q=!${bang}+${query}&format=json&pretty=1&no_redirect=1`
```

Comment:
Ditto quoting here. Even `bang` should be quoted, just so we can verify on this line alone that the query string can't be malformed.

Reply:
Done